six/modules/c++/six.sicd/source/ImageData.cpp

```diff
@@ -35,8 +35,21 @@ ImageData* ImageData::clone() const
     ImageData* d = new ImageData(*this);
     if (amplitudeTable)
     {
-	d->amplitudeTable = (AmplitudeTable*)amplitudeTable->clone();
+    d->amplitudeTable = (AmplitudeTable*)amplitudeTable->clone();
     }
     return d;
 }
+
+bool ImageData::operator==(const ImageData& rhs) const
+{
+    return (pixelType == rhs.pixelType &&
+        amplitudeTable == rhs.amplitudeTable &&

```

Comment:
This isn't quite right currently because `amplitudeTable` is a raw pointer.  Is there any good reason for it not to become a `ScopedCloneablePtr`?  Assuming that's ok, go ahead and make this change, remove the destructor and `clone()` method, and change `ComplexData` to hold a `ScopedCopyablePtr` of `ImageData`.  You'll probably need to ripple one or two things through the Python layer to keep it in sync too.